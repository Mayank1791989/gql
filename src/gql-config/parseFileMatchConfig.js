/* @flow */
import { type FileMatchConfig, type Globs } from './types';

export default function parseFileMatchConfig(
  match: FileMatchConfig,
): { glob: Globs, ignored: ?Globs } {
  // GlobPattern | Array<GlobPattern>
  if (typeof match === 'string' || Array.isArray(match)) {
    return { glob: match, ignored: [] };
  }

  return {
    glob: match.include,
    ignored: match.ignore || [],
  };
}
