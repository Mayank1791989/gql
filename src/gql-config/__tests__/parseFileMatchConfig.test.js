/* @flow */
import parseFileMatchConfig from '../parseFileMatchConfig';

test('string', () => {
  expect(parseFileMatchConfig('files/**/*.gql')).toMatchSnapshot();
});

test('Array<string>', () => {
  expect(
    parseFileMatchConfig(['files/**/*.gql', 'files2/**/*.gql']),
  ).toMatchSnapshot();
});

test('Object<include: string>', () => {
  expect(
    parseFileMatchConfig({
      include: 'files/**/*.gql',
    }),
  ).toMatchSnapshot();
});

test('Object<include: Array<string>>: single', () => {
  expect(
    parseFileMatchConfig({
      include: ['files/**/*.gql'],
    }),
  ).toMatchSnapshot();
});

test('Object<match: Array<string>>', () => {
  expect(
    parseFileMatchConfig({
      include: ['files/**/*.gql', 'files2/**/*gql'],
    }),
  ).toMatchSnapshot();
});

test('Object<include: Array<string>, ignore: string>', () => {
  expect(
    parseFileMatchConfig({
      include: ['files/**/*.gql', 'files2/**/*gql'],
      ignore: '**/test/**/*.gql',
    }),
  ).toMatchSnapshot();
});

test('Object<match: Array<string>, ignore: Array<string>>', () => {
  expect(
    parseFileMatchConfig({
      include: ['files/**/*.gql', 'files2/**/*gql'],
      ignore: ['**/test/**/*.gql'],
    }),
  ).toMatchSnapshot();
});
