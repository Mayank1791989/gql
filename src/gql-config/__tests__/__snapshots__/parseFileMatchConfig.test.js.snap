// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array<string> 1`] = `
Object {
  "glob": Array [
    "files/**/*.gql",
    "files2/**/*.gql",
  ],
  "ignored": Array [],
}
`;

exports[`Object<include: Array<string>, ignore: string> 1`] = `
Object {
  "glob": Array [
    "files/**/*.gql",
    "files2/**/*gql",
  ],
  "ignored": "**/test/**/*.gql",
}
`;

exports[`Object<include: Array<string>>: single 1`] = `
Object {
  "glob": Array [
    "files/**/*.gql",
  ],
  "ignored": Array [],
}
`;

exports[`Object<include: string> 1`] = `
Object {
  "glob": "files/**/*.gql",
  "ignored": Array [],
}
`;

exports[`Object<match: Array<string>, ignore: Array<string>> 1`] = `
Object {
  "glob": Array [
    "files/**/*.gql",
    "files2/**/*gql",
  ],
  "ignored": Array [
    "**/test/**/*.gql",
  ],
}
`;

exports[`Object<match: Array<string>> 1`] = `
Object {
  "glob": Array [
    "files/**/*.gql",
    "files2/**/*gql",
  ],
  "ignored": Array [],
}
`;

exports[`string 1`] = `
Object {
  "glob": "files/**/*.gql",
  "ignored": Array [],
}
`;
